packages/api/src/hooks/useTransactions.ts

```diff
@@ -48,8 +58,20 @@ const useTransactions = () => {
 
         return flatten_data?.map(transaction => ({
             ...transaction,
+            /** The transaction amount in currency format. */
+            display_amount: `${Intl.NumberFormat(preferred_language || 'en-US', {
+                minimumFractionDigits: fractional_digits,
+                maximumFractionDigits: fractional_digits,
+                minimumIntegerDigits: 1,
+            }).format(transaction?.amount || 0)} ${display_code}`,
```

Comment:
```suggestion
            display_amount: `${Intl.NumberFormat(preferred_language || 'en-US', {
                minimumFractionDigits: fractional_digits,
                maximumFractionDigits: fractional_digits,
                minimumIntegerDigits: 1,
            }).format(transaction?.amount || 0)} ${display_code}`,
```

`Intl.NumberFormat` will fail if a not null but invalid format preferred_language is sent by the backend. Please add error handling for this:

```        const displayBalanceFormat = (balance: number, currency: string) => {
            try {
                return `${Intl.NumberFormat(authorize_data?.preferred_language || 'en-US', {
                    minimumFractionDigits: 2,
                    maximumFractionDigits: 2,
                    minimumIntegerDigits: 1,
                }).format(balance)} ${currency || 'USD'}`;
            } catch (error) {
                return `${balance} ${currency || 'USD'}`;
            }
        };
        ````
        
        something like the snippets above

Reply:
Handled in https://github.com/binary-com/deriv-app/pull/11108